plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/hover/HoverProvider.java

```diff
@@ -105,4 +127,89 @@ private String renderContent(Hover hover) {
     }
     return Joiner.on("\n\n").join(contents);
   }
+
+  private void createHoverLinkListener(EditorPartPresenter activeEditor) {
+    final Window window = Elements.getWindow();
+    window.addEventListener(
+        Event.MOUSEDOWN,
+        evt -> {
+          Element anchorEle = (Element) evt.getTarget();
+
+          // Register the onClick and open only if the element is in a tooltip.
+          if (anchorEle.getOffsetParent() != null
+              && anchorEle.getOffsetParent().getClassList() != null
+              && anchorEle.getOffsetParent().getClassList().contains("textViewTooltipOnHover")
+              && anchorEle.hasAttribute("href")) {
+            String hrefContent = anchorEle.getAttribute("href");
+            Location uriLocation = getLocationFromURI(hrefContent);
+            anchorEle.setOnclick(
+                anchorEleClick -> {
+                  anchorEleClick.preventDefault();
+                  anchorEleClick.stopPropagation();
+                  if (hrefContent.startsWith("file:/") || hrefContent.startsWith("jdt:/")) {
+                    openFileInEditorHelper.openLocation(uriLocation);
+                  } else {
+                    BrowserUtils.openInNewTab(hrefContent);
+                  }
+                  ((TextEditor) activeEditor).getEditorWidget().hideTooltip();
+                });
+          }
+        });
+  }
+
+  /**
+   * Create a workable URI for opening files in the workspace
+   *
+   * @param uri The URI to fix
+   * @return A URI that can be opened by the editor
+   */
+  private String createWorkableURI(String uri) {
+    String fixedURI = uri;
+    if (uri.indexOf("#") != -1) {
+      fixedURI = fixedURI.substring(0, uri.lastIndexOf("#"));
+    }
+
+    // Because of the markdown2html library in jdt.ls file:///projects gets stripped to
```

Comment:
markdown2html? Shouldn't that be javadoc2markdown? If so, why don't we fix it in jdt.ls for all clients instead of handling it here?

Reply:
Yes I think the comment should say javadoc2markdown. The problem is that the library for javadoc2markdown always removes the two trailing slashes. So file:/// always gets changed to file:/. 

It looks like the javadoc2markdown library hasn't been worked on in years so i'm not sure how fast that change would be integrated/released in that and the only other option that I can think of is a do a string replace of all file:/ with file:///  in jdt.ls.

Reply:
Ok, I can live with that for now, but I think we should fix this, or all clients will have to work around it. Can you open a but against jdt.ls describing the problem?